_posts/2020-06-21-one-stop-guide-to-lombok.md

```diff
@@ -0,0 +1,337 @@
+---
+title: "One-stop Guide to Lombok"
+date: 2020-06-21 10:00 +0500
+modified: 2020-06-21 10:15 +0500
+---
+
+# What is it
+Lombok is a Java library which makes Java less chatty whilst making your code more readable. Java is great language but when you use its features you have to write a lot of boilerplate code.
+
+# Why to use Lombok
+
+Lombok helps you with a set of annotations to make your code more readable without losing its functionality. You don’t need to write typical constructors, setters or getters. With the help of Lombok you can easily add logging feature to you class or create fully functional builder.
+
+# Which the most popular features it has
+
+## @Getter and @Setter annotations
+These annotations generate setters and getters for a field respectively. These annotations can be applied as method level or class level.
+
+Lombok annotated code
+
+```java
+@Getter
+@Setter
+public class Person {
+  private String name;
+  private boolean employed;
+}
+```
+
+Generated Java code
```

Comment:
The "generated code" sections take up a lot of space in this article. Also they make this article feel like a copy of the Lombok docs. I would rather remove them and replace them with code that shows how to use the features (i.e. code that uses the setter and getter in this case). And this code should be pushed to the [code-examples](https://github.com/thombergs/code-examples/pulls) repo so that the readers can review it.

Reply:
Also, just listing "Lombok annotated code" and "Generated Java code" in each section is a little boring. Instead, discuss a feature and then introduce the code with a sentence like this.

With the `Bla` annotation we can do X:

```
code
```